casper/src/main/resources/PoS.rhox

```diff
@@ -311,39 +462,29 @@ in {
                                       new transferDoneCh in {
                                         @posVault!(
                                           "transfer",
-                                          coopRevAddress,
+                                          coopMultiVaultAddr,
                                           valBond,
                                           posAuthKey,
                                           *transferDoneCh
                                         ) |
-                                        for (_ <- transferDoneCh) {
-                                          // Moves slashed validator to withdrawers map to unbond at next epoch change.
-                                          new blockDataCh,
-                                              getBlockData(`rho:block:data`)
-                                          in {
-                                            getBlockData!(*blockDataCh) |
-                                            for (@blockNumber, _, _ <- blockDataCh) {
-                                              // Sets slashed validator's committed rewards and bond to 0 in state.
-                                              // Moves slashed validator to withdrawers map with no quarantine period.
-                                              stateUpdateCh!({
-                                                "allBonds"        : state.get("allBonds").set(validator, 0),
-                                                "activeValidators": state.get("activeValidators"),
-                                                "withdrawers"     : state.get("withdrawers")
-                                                                          .set(validator,
-                                                                            $$quarantineLength$$ + $$epochLength$$ * (1 + blockNumber / $$epochLength$$)),
-                                                "randomImages"    : state.get("randomImages"),
-                                                "randomNumbers"   : state.get("randomNumbers")
-                                              },
-                                              (true, Nil))
-                                            }
-                                          }
+                                        for (@(true, _) <- transferDoneCh) {
```

Comment:
It's definitely better to slash only if transfer is successful but now we don't handle failure. In case of transfer error slash should return error also.